src/server/api/routers/householdUser/householdUserRouter.ts

```diff
@@ -28,40 +41,203 @@ export const householdUserRouter = createTRPCRouter({
+          });
+        }
+
+        /**
+         * Check if invite has already been used and throw error if so
+         */
+        if (invite.status != "PENDING") {
+          throw new TRPCError({
+            code: "BAD_REQUEST",
+            message: `Invite with token ${inviteToken} already used`,
+          });
+        }
+
+        return invite;
+      } catch (e) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: `Invite with token ${inviteToken} not found`,
+          cause: e,
+        });
       }
-      // If userFoundByEmail then we can send an internal invite to the already registered user.
     }),
   /**
-   * Current assumption that the user to add is already a signed up user -
-   * Future Feature to develop - send email request to completely new user.
+   * Route to handle the result of the invite being accepted or declined.
+   * A user can either accept or decline an invite, via the link in the email or via the dashboard.
+   * The email link will direct to a route /accept-invite?inviteToken=xxxx - this route will call this procedure with the token and accepted (true/false).
+   * The dashboard will have a list of pending invites with accept/decline buttons - which will call this procedure with the userId and accepted.
+   * Input: userId?, inviteToken?, accepted: 'accept' | 'decline'
+   * This route will:
+   * 1. Validate the inviteToken or userId to find the invite record.
+   * 2. If accepted is 'accept', add the user to the household via addSingleUserToHousehold function.
+   * 3. Update the invite record with status 'accepted' or 'declined'.
+   * 4. Return the result.
    */
-  addNewUserToHousehold: protectedProcedure
-    .input(z.object({ householdId: z.string(), userEmail: z.string().email() }))
+  processUserHouseholdInvite: protectedProcedure
+    .input(
+      z.object({
+        inviteToken: z.string().min(1),
+        userId: z.string().min(1),
+        accepted: z.boolean(),
+      }),
+    )
     .mutation(async ({ ctx, input }) => {
-      const newMemberId = await findUserByEmail(input.userEmail, ctx.db);
-      if (!newMemberId) {
+      /**
+       * Find the invite record by either inviteToken or userId
+       * Throw error if not found
+       */
+      const inviteRecord = await ctx.db.householdInvite.findFirst({
+        where: input.inviteToken
+          ? { token: input.inviteToken }
+          : { invitedUserId: input.userId },
+      });
+      if (!inviteRecord) {
         throw new TRPCError({
           code: "NOT_FOUND",
-          message: `User with email ${input.userEmail} not found`,
+          message: `Invite not found`,
         });
       }
+
+      const { householdId, id } = inviteRecord;
+
       try {
-        const result = await addSingleUserToHousehold(
-          input.householdId,
-          newMemberId.id,
+        /**
+         * Add user to household and update invite status to ACCEPTED
```

Comment:
@AliMcG we are not catching a user that Declines the invite. It is adding a user to houselhold even if they decline.